app/controllers/users/edit_mobile_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Users
+  class EditMobileController < ApplicationController
+    include PhoneConfirmation
+
+    before_action :confirm_two_factor_authenticated
```

Comment:
In general I wonder if we should be using `prepend_before_action` here and elsewhere, but that would be a separate PR